docrawl/docrawl_core.py

```diff
@@ -149,20 +135,71 @@ def _initialise_browser(self):
                 window_size_x = 1450
 
             try:
-                self.browser = webdriver.Chrome(options=self.options, service=Service(ChromeDriverManager().install()), seleniumwire_options=sw_options)
+                self.browser = webdriver.Chrome(
+                    options=self.options, service=Service(ChromeDriverManager().install()),
+                    seleniumwire_options=sw_options
+                )
             except Exception as e:
                 docrawl_logger.error(f'Error while creating Chrome instance {e}')
                 self.browser = webdriver.Chrome(options=self.options)
 
         window_size_x = 1820
-
         self.browser.set_window_size(window_size_x, 980)
 
+        browser_meta_data['browser']['pid'] = self._determine_browser_pid()
+        if browser_meta_data.get('request'):
+            browser_meta_data['request']['loaded'] = False
+        self.docrawl_client.set_browser_meta_data(browser_meta_data)
+        docrawl_logger.info(f'Browser settings: {browser_meta_data}')
+
         return self.browser
 
+    def _close_browser(self, inp):
+        """
+        Close browser (remove driver instance).
+
+        :param browser: driver instance
+        """
+        try:
+            self.browser.quit()
+        except ConnectionRefusedError:
+            pass
+        except Exception as e:
+            docrawl_logger.error(f'Error while closing the browser: {e}')
+
+        # # Remove proxy after closing browser instance
```

Comment:
Proxy settings must be unset after browser is closed since they're stored within docrawl client instance an and thus will be reapplied again. Instead proxy should be explicitly set on each browser instance set up.

Reply:
I cannot unset proxy settings here as this method is used when restarting a browser after a crash and proxy settings might still need to be applicable. I've shifted unsetting proxy to `close_browser` in `docrawl_client.py`

Our 2 handlers - `OpenBrowser` and `CloseBrowser` are running all actions to open and close a new browser context - initializing new `browser_metadata`, setting `proxy` etc.